lib/_http_server.js

```diff
@@ -348,6 +349,18 @@ Server.prototype.setTimeout = function setTimeout(msecs, callback) {
   return this;
 };
 
+Server.prototype[EE.captureRejectionSymbol] = function(
+  err, event, req, res) {
+
+  switch (event) {
+    case 'request':
+      res.statusCode = 500;
+      res.end('something went wrong');
+      break;
```

Comment:
I think this should look something like:

```js
    if (!res.headersSent && !res.finished) {
      // Don't leak headers.
      for (const name of res.getHeaderNames()) {
        res.removeHeader(name)
      }
      res.statusCode = 500
      res.end()
    } else {
      res.destroy()
    }
```

Reply:
Good spot!